core/src/main/java/com/google/instrumentation/stats/Distribution.java

```diff
@@ -0,0 +1,236 @@
+      }
+      bucketCounts = new Long[bucketBoundaries.size() + 1];
+      for (int i = 0; i < bucketCounts.length; i++) {
+        bucketCounts[i] = 0L;
+      }
+    }
+  }
+
+  /**
+   * The number of values in the population. Must be non-negative.
+   *
+   * @return The number of values in the population.
+   */
+  public long getCount() {
+    return count;
+  }
+
+  /**
+   * The arithmetic mean of the values in the population. If {@link #getCount()} is zero then this
+   * value will be NaN.
+   *
+   * @return The arithmetic mean of all values in the population.
+   */
+  public double getMean() {
+    return sum / count;
+  }
+
+  /**
+   * The sum of the values in the population. If {@link #getCount()} is zero then this value must
+   * also be zero.
+   *
+   * @return The sum of values in the population.
+   */
+  public double getSum() {
+    return sum;
+  }
+
+  /**
+   * The range of the population values. If {@link #getCount()} is zero then this returned range is
+   * implementation-dependent. @ return The {code Range} representing the range of population
+   * boundaries.
+   */
+  public Range getRange() {
+    return range;
+  }
+
+  /**
+   * A Distribution may optionally contain a histogram of the values in the population. The
+   * histogram is given in {@link #getBucketCounts()} as counts of values that fall into one of a
+   * sequence of non-overlapping buckets, described by {@link
+   * DistributionAggregationDescriptor#getBucketBoundaries()}. The sum of the values in {@link
+   * #getBucketCounts()} must equal the value in {@link #getCount()}.
+   *
+   * <p>Bucket counts are given in order under the numbering scheme described above (the underflow
+   * bucket has number 0; the finite buckets, if any, have numbers 1 through N-2; the overflow
+   * bucket has number N-1).
+   *
+   * <p>The size of {@link #getBucketCounts()} must be no greater than N as defined in {@link
+   * DistributionAggregationDescriptor#getBucketBoundaries()}.
+   *
+   * <p>Any suffix of trailing buckets containing only zero may be omitted.
+   *
+   * <p>{@link #getBucketCounts()} will return null iff the associated {@link
+   * DistributionAggregationDescriptor#getBucketBoundaries()} returns null.
+   *
+   * @return The count of population boundaries in each histogram bucket.
+   */
+  @Nullable
+  public List<Long> getBucketCounts() {
+    if (hasBuckets()) {
+      return Arrays.asList(bucketCounts);
+    }
+    return null;
+  }
+
+  // Increment the appropriate bucket count. hasBuckets() MUST be true.
+  private void putIntoBucket(double value) {
+    for (int i = 0; i < bucketBoundaries.size(); i++) {
+      if (value < bucketBoundaries.get(i)) {
+        bucketCounts[i]++;
+        return;
+      }
+    }
+    bucketCounts[bucketCounts.length - 1]++;
+  }
+
+  /**
+   * Put a new value into the Distribution.
+   *
+   * @param value new value to be added to population
+   */
+  public void put(double value) {
```

Comment:
I have only seen "put" used for Java maps, so I think "add" would be a little clearer.